x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -733,70 +768,68 @@ export class ManifestManager {
    * @returns Artifact[]
    */
   private async listAllArtifacts(): Promise<Artifact[]> {
-    const fleetArtifacts = [];
-    const perPage = 100;
-    let page = 1;
+    const fleetArtifacts: Artifact[] = [];
+    let total = 0;
 
-    let fleetArtifactsResponse = await this.artifactClient.listArtifacts({
-      perPage,
-      page,
-    });
-    fleetArtifacts.push(...fleetArtifactsResponse.items);
-
-    while (
-      fleetArtifactsResponse.total > fleetArtifacts.length &&
-      !isEmpty(fleetArtifactsResponse.items)
-    ) {
-      page += 1;
-      fleetArtifactsResponse = await this.artifactClient.listArtifacts({
-        perPage,
-        page,
-      });
-      fleetArtifacts.push(...fleetArtifactsResponse.items);
+    for await (const artifacts of this.artifactClient.fetchAll()) {
+      fleetArtifacts.push(...artifacts);
+      total += artifacts.length;
     }
+
+    this.logger.info(`Count of current stored artifacts: ${total}`);
+
     return fleetArtifacts;
   }
 
   /**
-   * Cleanup .fleet-artifacts index if there are some orphan artifacts
+   * Pulls in all artifacts from Fleet and checks to ensure they are all being referenced
+   * by the Manifest. If any are found to not be in the current Manifest (orphan), they
+   * are cleaned up (deleted)
    */
   public async cleanup(manifest: Manifest) {
-    try {
-      const fleetArtifacts = await this.listAllArtifacts();
-      if (isEmpty(fleetArtifacts)) {
-        return;
-      }
-
-      const badArtifacts = [];
-      const badArtifactIds = [];
+    const badArtifactIds: string[] = [];
+    const errors: string[] = [];
+    const artifactDeletionProcess = new QueueProcessor<string>({
+      batchSize: this.packagerTaskPackagePolicyUpdateBatchSize,
```

Comment:
should we include new config value for it or just let the default (10)?

Reply:
Also, this is doing a bulk delete by id (if I'm not wrong), should it be a higher value?

Reply:
Yes, you are right. But it refers to package policy update batch size, and this area is about cleaning orphan artifacts, so maybe we can add a new config or let the default value.

Reply:
I thought about that, but figure we could just use the same one rather than introduce more config properties for batch size. I don't think it matters though. Do you have specific concerns?

if you rather have one, we can add it to the other issue we have to further brainstorm efficiencies for artifact creation, rather than to further delay this PR. Is that ok with you?